lua/autorun/misc.lua

```diff
@@ -67,4 +67,29 @@ else
 			net.SendToServer()
 		end
 	end
+	
+	hook.Add("PlayerDeath","DeathInPit",function(ply)
+		if IsValid(ply) then
+			local loc = ply:IsPlayer() and ply:GetLocation() or Location.Find(ply)
+			local name = Location.GetLocationNameByIndex(loc)
+			if name == "The Pit" then
+				ply.PitDeath = true
+			end
+		end
+	end)
+
+	hook.Add("PlayerSpawn","SpawnNextToPit",function(ply)

```

Comment:
The proper way to do this is with https://wiki.garrysmod.com/page/GM/PlayerSelectSpawn